contribs/gmf/src/directives/layertree.js

```diff
@@ -241,7 +242,9 @@ gmf.LayertreeController.prototype.getLayer = function(node, opt_depth,
     var isMerged = type === gmf.Themes.NodeType.NOT_MIXED_GROUP;
     layer.set('isMerged', isMerged);
 
-    this.dataLayerGroup_.getLayers().insertAt(0, layer);
+    if (!parentNode || gmf.Themes.getNodeType(parentNode) !== gmf.Themes.NodeType.MIXED_GROUP) {
```

Comment:
I think here, if the parent node is a mixed, then we should add the layer to parentNode.layer (group) instead of in the map. 
This way, everything is build on the fly, that would remove the logic from `getLayerCaseMixedGroup_`.